app/server/appsmith-server/src/main/java/com/appsmith/server/services/DatasourceServiceImpl.java

```diff
@@ -95,6 +95,11 @@ public DatasourceServiceImpl(Scheduler scheduler,
             return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.ORGANIZATION_ID));
         }
 
+        // If Authentication Details are present in the datasource, encrypt the details before saving
+        if (datasource.getDatasourceConfiguration() != null && datasource.getDatasourceConfiguration().getAuthentication() != null) {
+            datasource.getDatasourceConfiguration().setAuthentication(encryptAuthenticationFields(datasource.getDatasourceConfiguration().getAuthentication()));
+        }
```

Comment:
Since the `encryptAuthenticationFields` method modifies the authentication objects passed to it, we can simplify this to just:

```suggestion
        if (datasource.getDatasourceConfiguration() != null) {
            encryptAuthenticationFields(datasource.getDatasourceConfiguration().getAuthentication());
        }
```

And a null check inside the `encryptAuthenticationFields` method to make it null-tolerant.